plutus-core/plutus-ir/test/transform/inline/letApp

```diff
@@ -0,0 +1,47 @@
+(let
```

Comment:
This seems to be more lengthy than needed? Will look into this more. This is the PlutuxTx code:
```haskell
monoId :: CompiledCode Integer
{-# NOINLINE monoId #-}
monoId = plc (Proxy @"monoId") (
    let appNum :: Integer
        {-# NOINLINE appNum #-}
        appNum = 4
        funApp :: Integer -> Integer
        {-# NOINLINE funApp #-}
        funApp = (\x y -> x + y) appNum
    in funApp 5
    )
```

Reply:
Updated to a simplified program. See investigation [here](https://gist.github.com/thealmarty/e967958f8d4cfc3657e76fb0cf561425).